src/components/Datagrid/customCells/MoneyDiscountedCell.tsx

```diff
@@ -0,0 +1,133 @@
+import OrderDiscountCommonModal from "@dashboard/orders/components/OrderDiscountCommonModal/OrderDiscountCommonModal";
+import {
+  ORDER_LINE_DISCOUNT,
+  OrderDiscountCommonInput,
+} from "@dashboard/orders/components/OrderDiscountCommonModal/types";
+import { useOrderLineDiscountContext } from "@dashboard/products/components/OrderDiscountProviders/OrderLineDiscountProvider";
+import {
+  CustomCell,
+  CustomRenderer,
+  getMiddleCenterBias,
+  GridCellKind,
+} from "@glideapps/glide-data-grid";
+import React, { useCallback } from "react";
+
+interface MoneyDiscountedCellProps {
+  readonly kind: "money-discounted-cell";
+  readonly currency: string;
+  readonly undiscounted?: string | number;
+  readonly value: number | string | null;
+  readonly lineItemId?: string;
+  readonly locale: Locale;
+}
+
+export type MoneyDiscuntedCell = CustomCell<MoneyDiscountedCellProps>;
+
+const MoneyDiscountedCellEditor = ({ onFinishedEditing, value }) => {
+  const getDiscountProviderValues = useOrderLineDiscountContext();
+  const editedLineId = value.data.lineItemId;
+  const discountProviderValues = editedLineId
+    ? getDiscountProviderValues(editedLineId)
+    : null;
+
+  const handleDiscountConfirm = useCallback(
+    async (discount: OrderDiscountCommonInput) => {
+      await discountProviderValues.addOrderLineDiscount(discount);
+      onFinishedEditing(undefined);
+    },
+    [discountProviderValues, onFinishedEditing],
+  );
+  const handleDiscountRemove = useCallback(async () => {
+    await discountProviderValues.removeOrderLineDiscount();
+    onFinishedEditing(undefined);
+  }, [discountProviderValues, onFinishedEditing]);
+
+  return (
+    <OrderDiscountCommonModal
+      onClose={() => onFinishedEditing(undefined)}
+      modalType={ORDER_LINE_DISCOUNT}
+      maxPrice={discountProviderValues.unitUndiscountedPrice}
+      onConfirm={handleDiscountConfirm}
+      onRemove={handleDiscountRemove}
+      existingDiscount={discountProviderValues.orderLineDiscount}
+      confirmStatus={discountProviderValues.orderLineDiscountUpdateStatus}
+      removeStatus={discountProviderValues.orderLineDiscountRemoveStatus}
+    />
+  );
+};
+
+export const moneyDiscountedCellRenderer =
+  (): CustomRenderer<MoneyDiscuntedCell> => ({
+    kind: GridCellKind.Custom,
+    isMatch: (c): c is MoneyDiscuntedCell =>
+      (c.data as any).kind === "money-discounted-cell",
+    draw: (args, cell) => {
+      const { ctx, theme, rect } = args;
+      const { currency, value, undiscounted, locale } = cell.data;
+      const hasValue = value === 0 ? true : !!value;
+      const formattedValue = value
+        ? formatMoneyAmount({ amount: Number(value), currency }, locale)
+        : "-";
+
+      const formattedUndiscounted =
+        undiscounted && undiscounted !== value
+          ? formatMoneyAmount(
+              { amount: Number(undiscounted), currency },
+              locale,
+            )
+          : "";
```

Comment:
**issue**: this code is copy-pasted as well. If you create two cells that are similar to each other, create some base ona and use composition, try to never copy-paste the code when it's contextually connected to each other. In case you have reason for it, please leave a comment